src/shared/components/views/DatatableWithToggle.tsx

```diff
@@ -20,29 +25,37 @@ const DatatableWithToggle = ({ table }: { table: JSX.Element }) => {
     'protvista-datatable'
   );
 
-  const toggleTableState = async () => {
-    const tableEl = tableRef?.current;
-    if (tableEl) {
-      if (tableState === 'Expand') {
-        tableEl.expandTable = true;
-        setTableState('Collapse');
-      } else {
-        tableEl.expandTable = false;
-        setTableState('Expand');
-        await frame();
-        tableContainerRef?.current?.scrollIntoView({
+  // On expand/collapse change
+  useEffect(() => {
+    // except on first render
+    if (!firstRenderRef.current) {
+      // Scroll table back into view when collapsing
+      if (!expandTable) {
+        tableRef.current?.parentElement?.scrollIntoView({
           behavior: 'smooth',
           block: 'center',
         });
       }
+      gtagFn('event', 'feature viewer', {
+        event_category: `datatable ${expandTable ? 'expanded' : 'collapsed'}`,
+        event_label: params.accession,
+      });
     }
-  };
+    // If first render was previous render, then it's not the first anymore...
+    if (firstRenderRef.current) {
+      firstRenderRef.current = false;
+    }
+  }, [expandTable, params.accession]);
 
   if (datatableElement.defined) {
     return (
-      <div ref={tableContainerRef} className="datatable-with-toggle">
-        <datatableElement.name ref={tableRef} filter-scroll>
-          {table}
+      <div className="datatable-with-toggle">
+        <datatableElement.name
+          ref={tableRef}
+          filter-scroll
+          expandtable={expandTable ? '' : undefined}
```

Comment:
Since React doesn't support web components too well, if I were to use the boolean it would be rendered as a string as `expandtable="true"` or `expandtable="false"` (truthy anyway), but in our custom element logic we tend to follow the usual pattern of "attribute present means true" vs "attribute absent means false". So here I used "undefined" to tell React to remove the attribute